Facelift, support for py3 and modern django version, wheel support.

EnTeQuAk opened the pull request:
* Add support for Python 3 and PyPy
* Move to pytest for testing
* Add wheel build support
* Drops support for Django < 1.6, adds support for Django 1.6, 1.7 1.8 and 1.9

This adds a proper tox config and uses this in travis, moves to pytest
instead of nose and cleans up some old stuff.

muffinresearch commented:
Looks like this needs a docs update for how to run the tests. See https://django-csp.readthedocs.org/en/latest/contributing.html#tests

muffinresearch commented on setup.py:
```diff
@@ -1,29 +1,75 @@
-    author='James Socol',
-    author_email='me@jamessocol.com',
+    long_description=read('README.rst'),
+    author='Christopher Grebs',
```
Could maybe use `maintainer` and `maintainer_email` fields instead of updating the author fields?

EnTeQuAk replied:
you're right, that makes sense. I always forget about these fields. fixed in d8a5d40

EnTeQuAk commented:
Great catch, I updated the docs to reference py.test in d8a5d40

muffinresearch commented:
> Great catch, I updated the docs to reference py.test in d8a5d40

Looks like there's some docs referencing the `requirements.txt` which is being removed in this PR.

EnTeQuAk commented:
Fixed too :)
 
 
On Wed, Apr 27, 2016, at 05:57 PM, Stuart Colville wrote:
>> Great catch, I updated the docs to reference py.test in d8a5d40[1]
> Looks like there's some docs referencing the requirements.txt which is
> being removed in this PR.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub[2]
>

Links:

  1. https://github.com/mozilla/django-csp/commit/d8a5d40939e8d2338b3da0c183318c0516b3553c
  2. https://github.com/mozilla/django-csp/pull/61#issuecomment-215130002
